cpp/include/raft/linalg/detail/gemm.hpp

```diff
@@ -13,231 +13,135 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 #pragma once
 
-#include <cublas_v2.h>
-
-#include "cublas_wrappers.hpp"
+#include "cublaslt_wrappers.hpp"
 
-#include <raft/core/resource/cublas_handle.hpp>
 #include <raft/core/resources.hpp>
 
-namespace raft {
-namespace linalg {
-namespace detail {
+namespace raft::linalg::detail {
 
-/**
- * @brief the wrapper of cublas gemm function
- *  It computes the following equation: C = alpha .* opA(A) * opB(B) + beta .* C
- *
- * @tparam math_t the element type
- * @tparam DevicePointerMode whether pointers alpha, beta point to device memory
- * @param [in] handle raft handle
- * @param [in] trans_a cublas transpose op for A
- * @param [in] trans_b cublas transpose op for B
- * @param [in] m number of rows of C
- * @param [in] n number of columns of C
- * @param [in] k number of rows of opB(B) / number of columns of opA(A)
- * @param [in] alpha host or device scalar
- * @param [in] A such a matrix that the shape of column-major opA(A) is [m, k]
- * @param [in] lda leading dimension of A
- * @param [in] B such a matrix that the shape of column-major opA(B) is [k, n]
- * @param [in] ldb leading dimension of B
- * @param [in] beta host or device scalar
- * @param [inout] C column-major matrix of size [m, n]
- * @param [in] ldc leading dimension of C
- * @param [in] stream
- */
-template <typename math_t, bool DevicePointerMode = false>
-void gemm(raft::resources const& handle,
-          const bool trans_a,
-          const bool trans_b,
-          const int m,
-          const int n,
-          const int k,
-          const math_t* alpha,
-          const math_t* A,
-          const int lda,
-          const math_t* B,
-          const int ldb,
-          const math_t* beta,
-          math_t* C,
-          const int ldc,
-          cudaStream_t stream)
+template <typename T, bool DevicePointerMode = false>
+void legacy_gemm(raft::resources const& res,
```

Comment:
Same here- cublaslt handle + stream or raft resources but not both.

Reply:
https://github.com/rapidsai/raft/pull/1736#discussion_r1452113436